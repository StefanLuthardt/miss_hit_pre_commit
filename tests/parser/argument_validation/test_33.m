% Contributed on #60 after it was closed

function out = my_function(A, B, C)
    arguments
        A (1, 1) string {}
        B (1, :) double {}
        C (1, 4) cell {}
    end
end
