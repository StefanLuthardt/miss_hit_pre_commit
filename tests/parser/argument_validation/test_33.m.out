----------------------------------------------------------------------
--  Parse tree for test_33.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <my_function>
      Inputs: Identifier <A>
      Inputs: Identifier <B>
      Inputs: Identifier <C>
      Outputs: Identifier <out>
    Validation: Arguments Special_Block
      Items: Entity_Constraints
        Dimension 1 constraint: 1
        Dimension 2 constraint: 1
        Name: Identifier <A>
        Class: Identifier <string>
      Items: Entity_Constraints
        Dimension 1 constraint: 1
        Dimension 2 constraint: COLON
        Name: Identifier <B>
        Class: Identifier <double>
      Items: Entity_Constraints
        Dimension 1 constraint: 1
        Dimension 2 constraint: 4
        Name: Identifier <C>
        Class: Identifier <cell>
    Body: Sequence_Of_Statements
----------------------------------------------------------------------
MISS_HIT Debug Summary: 1 file(s) analysed, everything seems fine
